multiqc/modules/base_module.py

```diff
@@ -281,14 +281,70 @@ def add_section(
             }
         )
 
-    def clean_s_name(self, s_name, f=None, root=None, filename=None, seach_pattern_key=None):
-        """Helper function to take a long file name and strip it
-        back to a clean sample name. Somewhat arbitrary.
-        :param s_name: The sample name to clean
+    @staticmethod
+    def _clean_fastq_name_pair(clean_names: List[str]) -> str:
+        """
+        Extract a sample name from a list of file names - for example, FASTQ pairs.
+        """
+        if len(set(clean_names)) == 1:  # all the same
+            return clean_names[0]
+
+        if len(clean_names) == 2:
+            # Possibly a FASTQ pair - try trimming FASTQ suffixes.
+            r1, r2 = sorted(clean_names)
+
+            # Try trimming the conventional illumina suffix with a tail 001 ending. Refs:
+            # https://support.illumina.com/help/BaseSpace_Sequence_Hub_OLH_009008_2/Source/Informatics/BS/NamingConvention_FASTQ-files-swBS.htm
+            # https://support.10xgenomics.com/spatial-gene-expression/software/pipelines/latest/using/fastq-input#:~:text=10x%20pipelines%20need%20files%20named,individual%20who%20demultiplexed%20your%20flowcell.
+            cleaned_r1 = re.sub(r"_R1_\d\d\d$", "", r1)
+            cleaned_r2 = re.sub(r"_R2_\d\d\d$", "", r2)
```

Comment:
```suggestion
            cleaned_r1 = re.sub(r"_R1_\d{3}$", "", r1)
            cleaned_r2 = re.sub(r"_R2_\d{3}$", "", r2)
```

Reply:
I personally prefer to duplicate, but that's alright :)